apache-poi-3/src/test/java/com/baeldung/fastexcel/FastexcelDetectEmptyRowUnitTest.java

```diff
@@ -0,0 +1,45 @@
+package com.baeldung.fastexcel;
+
+import static org.junit.Assert.assertTrue;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.stream.Stream;
+
+import org.dhatim.fastexcel.reader.ReadableWorkbook;
+import org.dhatim.fastexcel.reader.Row;
+import org.dhatim.fastexcel.reader.Sheet;
+import org.junit.Before;
+import org.junit.Test;
+
+
+public class FastexcelDetectEmptyRowUnitTest {
+
+    private FastexcelHelper fastexcelHelper;
+    private static String FILE_NAME = "consumer_info_with_empty_row.xlsx";
+    private String fileLocation;
+
+    @Before
+    public void loadExcelFile() throws IOException {
+        fileLocation = "src/main/resources/" + FILE_NAME;
+        fastexcelHelper = new FastexcelHelper();
+    }
+
+    @Test
+    public void whenParsingFastExcelFile_thenDetectEmptyRow() throws IOException {
+        try (FileInputStream file = new FileInputStream(fileLocation); ReadableWorkbook wb = new ReadableWorkbook(file)) {
+            Sheet sheet = wb.getFirstSheet();
+            try (Stream<Row> rows = sheet.openStream()) {
```

Comment:
Why is this implementation not possible?

```java
// assert all rows are empty
try (Stream<Row> rows = sheet.openStream()) {
    boolean isEmpty = rows.allMatch(fastexcelHelper::isRowEmpty);

    assertTrue(isEmpty);
}

// assert last row is empty
try (Stream<Row> rows = sheet.openStream()) {
    Optional<Row> last = rows.reduce((first, second) -> second);
    boolean isLastRowEmpty = last.map(fastexcelHelper::isRowEmpty).orElseThrow();
    assertTrue(isLastRowEmpty);
}
```

Reply:
Done, I added test cases for an empty file and a file with a few records but empty rows.